File: Grass.cpp
#include "include/Grass.h"

Grass::Grass(World* world, Point& Position)
    : Plant(world, 0, position, "🌱", 0) {

}

Grass::Grass(World* world, Point& position, int age)
    : Plant(world, 0, position, "🌱", age) {

}

Grass::~Grass() {
    // Destructor implementation
}

File: game-of-life-java/src/Grass.java
import javax.swing.*;
import java.awt.*;

public class Grass extends Plant {
    private static final ImageIcon grassIcon = new ImageIcon("images/grass.png");
    static final Image scaledGrassIcon = grassIcon.getImage().getScaledInstance(30, 30, Image.SCALE_SMOOTH);

    public Grass(Point position, World world, int age) {
        super(0, position, world, age);
    }

    @Override
    public ImageIcon getIcon() {
        return grassIcon;
    }
}
